servatrice/src/serversocketinterface.cpp

```diff
@@ -946,14 +960,21 @@ Response::ResponseCode AbstractServerSocketInterface::cmdRegisterAccount(const C
         {
             QSqlQuery *query = sqlInterface->prepareQuery("insert into {prefix}_activation_emails (name) values(:name)");
             query->bindValue(":name", userName);
-            if (!sqlInterface->execSqlQuery(query))
+            if (!sqlInterface->execSqlQuery(query)) {
+                rc.setResponseExtension(re);
                 return Response::RespRegistrationFailed;
+            }
 
+            re->set_custom_reason_str(servatrice->getCustomRegSuccessMessage().toStdString());
+            rc.setResponseExtension(re);
```

Comment:
This is the dirtiness factor I was referring to.  Since the response extension is created prior to the switch statement and the custom_reason_str gets set based on the results of the various tree outcomes you have to set the response code prior to returning the response.  I could possibly place a variable in the function that defines the type of response needs to be returned and set the response and send it below the switch statement.  Either way doesn't seem ideal to me.  I went this route as its seemed easier to understand through the various if tree's on were the code logic should end with a return for what the results being returned would be. (if that makes any since).